### PR TITLE
Centralize freestanding store helpers

### PR DESCRIPTION
**Pull request by teh-cmc**
Just a trivial PR to centralize a bunch of freestanding helpers and improve discoverability / autocomplete etc.

Related to #2019 (if we're very generous)

PR Build Summary: https://build.rerun.io/pr/2153

### CODE REVIEW

File: crates/re_arrow_store/src/store_helpers.rs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                    "Couldn't deserialize component at {entity_path}.{}: {err}",
+                    C::name()
+                );
+            })
+            .ok()?;
+
+        let component = iter.next();
+
+        if iter.next().is_some() {
+            re_log::warn_once!("Unexpected batch for {} at: {}", C::name(), entity_path);
```
Comment (teh-cmc) [inline crates/re_arrow_store/src/store_helpers.rs:45]: formatting

File: crates/re_query/src/query.rs [hunk 2]
```diff
@@ -79,7 +79,7 @@ pub fn get_component_with_instances(
 /// length.
 ///
 /// If you expect only one instance (e.g. for mono-components like `Transform` `Tensor`]
-/// and have no additional components you can use [`re_data_store::query_latest_single`] instead.
+/// and have no additional components you can use [`re_data_store::query_latest_component`] instead.
```
Comment (teh-cmc) [inline crates/re_query/src/query.rs:82]: xxx

**Review by jleibs — CHANGES_REQUESTED** [review]
Can we avoid conflicting with: https://github.com/rerun-io/rerun/pull/2010/files#diff-28ad8e81ea9af25145d4379530f89ee0c70da10021fa4b210680f0df2b3e01d3

**Comment by teh-cmc** [comment]
> Can we avoid conflicting with: https://github.com/rerun-io/rerun/pull/2010/files#diff-28ad8e81ea9af25145d4379530f89ee0c70da10021fa4b210680f0df2b3e01d3

I definitely don't mind handling the annoying rebase if needed

**Pull request merged**